### PR TITLE
designate: move keystone resource lookup to convergence (SOC-10887)

### PR DESCRIPTION
**Pull request by guangyee**
To avoid weird problems such as race conditions due to
changing Keystone endpoint protocols, admin password, etc.

### CODE REVIEW

File: chef/cookbooks/designate/recipes/common.rb [hunk 1]
```diff
@@ -36,21 +36,23 @@
 
 resource_project_id = ""
 keystone_settings = KeystoneHelper.keystone_settings(node, :designate)
-env = {
```
Comment (stefannica) [inline chef/cookbooks/designate/recipes/common.rb]: This is just to confirm something I had some doubts about earlier: it wasn't also necessary to also move this variable into the ruby block, running the openstack command there is enough.

**Comment by stefannica** [comment]
@guangyee by the way, you need a Jira ticket number to make Travis happy. Could you please create one to track this issue ?

File: chef/cookbooks/designate/recipes/common.rb (hunk 1 above)
Comment (nicolasbock) [reply chef/cookbooks/designate/recipes/common.rb]: I don't think I understand what you are trying to say :smile:

**Comment by guangyee** [comment]
> @guangyee by the way, you need a Jira ticket number to make Travis happy. Could you please create one to track this issue ?

Can you reuse SOC-10887?

**Comment by nicolasbock** [comment]
@gosipyan Could you check this PR?

**Comment by guangyee** [comment]
> i have triggered job and apply patch manually it failed during test setup
> https://ci.suse.de/blue/organizations/jenkins/openstack-crowbar/detail/openstack-crowbar/526/pipeline/139/

Looks like the dns-server role need to be added to the controller as well.

https://github.com/crowbar/crowbar-openstack/blob/master/chef/cookbooks/designate/recipes/mdns.rb#L21-#L28
Otherwise, we don't have any backend configured.

@gosipyan can you please reapply the DNS barclamp with dns-server added to the controller?

**Pull request merged**